src/resources/js/parts-kit.js

```diff
@@ -181,6 +133,141 @@ class Toggle {
     };
 }
 
-// Init the toggle class
-const toggles = document.querySelectorAll("[data-parts-kit-toggle]");
-toggles.forEach((el) => new Toggle(el));
+// Parts kit class for expanding/collapsing sidebar
+class PartsKit {
+    constructor(el) {
+        this.el = el;
+        this.setVars();
+        this.bindEvents();
+        this.initSidebar();
+    }
+
+    setVars() {
+        this.sidebar = this.el.querySelector("[data-sidebar]");
+        this.lsSidebar = "pk-sidebar-hidden";
+        this.fullscreenExpandBtn = this.el.querySelector(
+            "[data-fullscreen-expand]"
+        );
+        this.fullscreenCloseBtn = this.el.querySelector(
+            "[data-fullscreen-close]"
+        );
+
+        // Set up toggle classes
+        const toggleElements = document.querySelectorAll("[data-parts-kit-toggle]");
+        this.toggles = [...toggleElements].map(el => new Toggle(el));
+
+        /** @type {HTMLInputElement} */
+        this.sidebarSearch = this.el.querySelector("[data-sidebar-search]");
+        /** @type {NodeListOf<HTMLElement>} */
+        this.sectionItems = this.el.querySelectorAll("[data-section-item]");
+        this.searchList = [...this.sectionItems].map(el => {
+            const toggleEl = el.closest("[data-parts-kit-toggle]");
+            return {
+                el,
+                searchText: el.getAttribute('data-search-text')?.toLowerCase(),
+                toggleEl,
+                toggle: this.toggles.find(x => x.el === toggleEl)
+            }
+        });
+    }
+
+    bindEvents() {
+        document.addEventListener("keydown", this.handleKeydown);
+        this.fullscreenExpandBtn.addEventListener("click", this.toggleSidebar);
```

Comment:
Should these event listeners be removed at some point?

Reply:
This parts kit are just unique Craft CMS urls, so this JS is only called once on page creation. No real point where these events need to be removed.